File: Microservices/auth-service/service/LoginDetailsService.go
package service

import (
	"auth-service/model"
	"auth-service/repository"
)

type LoginDetailsService struct{
	LoginDetailsRepository *repository.LoginDetailsRepository
}

func (service *LoginDetailsService) Update(loginDetails *model.LoginDetails) error {
	err := service.LoginDetailsRepository.Update(loginDetails)
	if err != nil {
		return err
	}
	return nil
}
func (service *LoginDetailsService) Create(loginDetails *model.LoginDetails) error {
	service.LoginDetailsRepository.Create(loginDetails)
	return nil
}

func (service *LoginDetailsService) GetAll() []model.LoginDetails {
	return service.LoginDetailsRepository.GetAll()
}

func (service *LoginDetailsService) GetByEmail(email string) model.LoginDetails {
	return service.LoginDetailsRepository.GetByEmail(email)
}

File: Microservices/auth-service/src/main/java/rs/ac/uns/ftn/xws/team22/auth/service/impl/LoginDetailsService.java
package rs.ac.uns.ftn.xws.team22.auth.service.impl;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;
import rs.ac.uns.ftn.xws.team22.auth.model.LoginDetails;
import rs.ac.uns.ftn.xws.team22.auth.repository.LoginDetailsRepository;
import rs.ac.uns.ftn.xws.team22.auth.service.ILoginDetailsService;

import java.util.List;
import java.util.UUID;

@Service
public class LoginDetailsService implements ILoginDetailsService, UserDetailsService {

    @Autowired
    private LoginDetailsRepository loginDetailsRepository;

    @Override
    public List<LoginDetails> findAll() {
        return this.loginDetailsRepository.findAll();
    }

    @Override
    public LoginDetails findById(UUID id) {
        return this.loginDetailsRepository.getById(id);
    }

    @Override
    public LoginDetails findByEmail(String email) {
        return this.loginDetailsRepository.findByEmail(email);
    }

    @Override
    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
        return this.loginDetailsRepository.findByEmail(s);
    }
}
